src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -388,6 +393,15 @@ class MongoDBEventStoreImplementation implements MongoDBEventStore, Closeable {
       );
     }
 
+    await tryPublishMessagesAfterCommit<MongoDBEventStore>(
+      // @ts-expect-error Issues with `globalPosition` not being present causing the type for metadata to expect `never`
```

Comment:
@alex-laycalvert, that's weird; I noticed that TS got crazy on my other computer, but when I reinstalled fresh packages, then it worked fine. I'm not sure if that's the same issue, but I'll check the typing after the merge :+1:

Reply:
Yes, because technically, it's not `undefined` but optional. TypeScript is treating those cases differently. I'll have a look, it might be that I messed something out.